src/main/java/com/faforever/client/discord/DiscordRichPresenceService.java

```diff
@@ -0,0 +1,110 @@
+  public static final String SPECTATE_SECRET = GAME_ID_REGEX_NAME + "=(?<" + GAME_ID_REGEX_NAME + ">\\d*);" + PLAYER_ID_REGEX_NAME + "=(?<" + PLAYER_ID_REGEX_NAME + ">\\d*)";
+  public static final String JOIN_SECRET = GAME_ID_REGEX_NAME + "=(?<" + GAME_ID_REGEX_NAME + ">\\d*)";
+  private static final String HOSTING = "Hosting";
+  private static final String WAITING = "Waiting";
+  private static final String PLAYING = "Playing";
+  /**
+   * It is suggested(libaries github page) to look for callbacks every 5 seconds
+   */
+  private static final int INITIAL_DELAY_FOR_CALLBACK = 5000;
+  private static final int PERIOD_FOR_CALLBACK = 5000;
+
+  private final ClientProperties clientProperties;
+  private final PlayerService playerService;
+  private final Timer timer;
+
+
+  public DiscordRichPresenceService(PlayerService playerService, ClientDiscordEventHandler discordEventHandler, ClientProperties clientProperties) {
+    this.playerService = playerService;
+    this.clientProperties = clientProperties;
+    this.timer = new Timer(true);
+    try {
+      DiscordRPC.discordInitialize(this.clientProperties.getDiscordConfig().getApplicationId(), discordEventHandler, true);
+      timer.schedule(new TimerTask() {
+        @Override
+        public void run() {
+          DiscordRPC.discordRunCallbacks();
+        }
+      }, INITIAL_DELAY_FOR_CALLBACK, PERIOD_FOR_CALLBACK);
+    } catch (Exception e) {
+      //TODO: report to bugsnap
+      log.error("Error in discord init", e);
+    }
+  }
+
+  public void updatePlayedGameTo(Game game, int currentPlayerId) {
+    try {
+      if (game.getStatus() == GameStatus.CLOSED) {
+        DiscordRPC.discordClearPresence();
+      }
+      DiscordRichPresence.Builder discordRichPresence = new DiscordRichPresence.Builder(getDiscordState(game));
+      discordRichPresence.setDetails(MessageFormat.format("{0} | {1}", game.getFeaturedMod(), game.getTitle()));
+      discordRichPresence.setParty(String.valueOf(game.getId()), game.getNumPlayers(), game.getMaxPlayers());
+      discordRichPresence.setSmallImage(clientProperties.getDiscordConfig().getSmallImageKey(), "");
+      discordRichPresence.setBigImage(clientProperties.getDiscordConfig().getBigImageKey(), "");
+      String joinSecret = null;
+      String spectateSecrete = null;
+      if (game.getStatus() == GameStatus.OPEN) {
+        joinSecret = JOIN_SECRET.replaceAll("\\(?<gameId>(.*)\\)", String.valueOf(game.getId()));
+      }
+
+      if (game.getStatus() == GameStatus.PLAYING) {
+        spectateSecrete = SPECTATE_SECRET.replaceAll("\\(?<gameId>(.*)\\)", String.valueOf(game.getId()))
+            .replaceAll("\\(\\?<playerId>(.*)\\)", String.valueOf(currentPlayerId));
+      }
+
+      discordRichPresence.setSecrets(joinSecret, spectateSecrete);
+
+      DiscordRichPresence update = discordRichPresence.build();
+      if (game.getStartTime() != null) {
+        //sadly that can not be set via the Builder, I made an issue in the library's repo
```

Comment:
I don't want to be nit-picky but you already know that I am :D So this should read:
```
// Sadly, this can not be set via the builder. I created an issue in the library's repo
```
And then, it'd be very nice to have the link to the issue as well so we don't have to search it on our own.
Even better, a 
```
// TODO do <what> when <issue link> has been fixed
```